core-kotlin-modules/core-kotlin-10/src/test/kotlin/com/baeldung/cloningobject/CloningObjectUnitTest.kt

```diff
@@ -0,0 +1,74 @@
+package com.baeldung.cloningobject
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+data class Address(var street: String, var city: String)
+
+data class Person(var name: String, var address: Address) : Cloneable {
+    public override fun clone(): Person {
+        return Person(name, address)
+    }
+
+    fun deepCopy(name: String = this.name, address: Address = this.address.copy()): Person {
+        return Person(name, address)
+    }
+
+    constructor(person: Person) : this(person.name, person.address)
+}
+
+class CloningObjectUnitTest {
+
+    private val address = Address("Jln. Kemasan", "Yogyakarta")
+    private val person = Person("Hangga", address)
+
+    @Test
+    fun `when using secondary constructor then proves that shallow copy`() {
+        val clonedPerson = Person(person)
+        clonedPerson.address.city = "Surabaya"
+        clonedPerson.address.street = "Jln. Ahmad Yani"
```

Comment:
I think it would be clearer if we modified the original person rather than the cloned person in these tests, and showed the impact that does or doesn't have.

I also think it would be clearer if we did the isEqualTo using hard coded string constants rather than field references. This would also remove the need for the println statements, since we would be able to see the expected strings already in the test.

Let me know what you think - let's do a round on just the PR first before going back to the article, to avoid rework.

Reply:
Oh yeah, I agree.
that makes more sense.
I have done it just now. Please review again.
next, I'll update the article.